### PR TITLE
Restructure OAV device organisation and naming (Artemis554)

### PR DESCRIPTION
**Pull request by dperl-dls**
Goes with changes in  https://github.com/DiamondLightSource/python-artemis/pull/584

### CODE REVIEW

**Comment by DominicOram** [comment]
Can you make the PR title a bit more descriptive, dodal will use it for release note generation so should probably use something that makes sense to general DLS devs

File: src/dodal/devices/areadetector/plugins/MXSC.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class MXSC(Device):
+    """
+    Device for edge detection plugin.
+    """
+
+    input_plugin_pv: EpicsSignal = Component(EpicsSignal, "NDArrayPort")
+    enable_callbacks_pv: EpicsSignal = Component(EpicsSignal, "EnableCallbacks")
+    min_callback_time_pv: EpicsSignal = Component(EpicsSignal, "MinCallbackTime")
+    blocking_callbacks_pv: EpicsSignal = Component(EpicsSignal, "BlockingCallbacks")
```
Comment (DominicOram) [inline src/dodal/devices/areadetector/plugins/MXSC.py:12]: Could: I think the word PV is a bit redundant here. We know they're all PVs

**Review by DominicOram — APPROVED** [review]
Changes here look mostly good. Most comments are on Artemis PR. Do we still want the component in the OAV to be called snapshot, after renaming it to `MJPG`?

**Review by DominicOram — CHANGES_REQUESTED** [review]
Thanks, mostly looks good, couple of questions in code

File: .github/workflows/code.yml [hunk 2]
```diff
@@ -35,12 +35,12 @@ jobs:
       fail-fast: false
       matrix:
         os: ["ubuntu-latest"] # can add windows-latest, macos-latest
-        python: ["3.9", "3.10", "3.11"]
```
[the remark this answers is not in the document]
Comment (dperl-dls) [reply .github/workflows/code.yml:38]: 3.9 is still there, in the next bit which tests non-editable installation, I just kept the same format as before :)

[the remark this answers is not in the document]
Comment (DominicOram) [reply .github/workflows/code.yml:38]: Yes, I see, apologies

**Review by DominicOram — APPROVED** [review]
Great, thank you!

**Pull request merged**